src/render/Screen.java

```diff
@@ -0,0 +1,79 @@
+package traingame.render;
+
+import java.util.*;
+
+import traingame.engine.render.gui.*;
+import traingame.engine.render.text.BitmapFont;
+import traingame.Game;
+
+public abstract class Screen {
+    final int INVALID = -1;
+
+    ArrayList<GuiElement> selectable = new ArrayList<>();
+    int selected = INVALID;
+
+    protected void select(GuiElement element) {
+        for (int i = 0; i < selectable.size(); ++i) {
+            if (selectable.get(i).equals(element)) {
+                selected = i;
+            }
+        }
+    }
+
+    public void positionPointer(double xPos, double yPos) {
+        for (int i = 0; i < selectable.size(); ++i) {
+            if (selectable.get(i).containsPoint(xPos, yPos)) {
+                selected = i;
+            }
+        }
+    }
+
+    public GuiElement getSelected() {
+        if (selected == INVALID) {
+            return null;
+        }
+        return selectable.get(selected);
+    }
+
+    public boolean trigger() {
+        GuiElement selected = getSelected();
+        if (selected != null) {
+            return selected.trigger();
+        }
+        return false;
+    }
+
+    public boolean click(double xPos, double yPos) {
+        positionPointer(xPos, yPos);
+        return trigger();
+    }
+
+    public void up() {
+        if (selected > 0) {
+            selected -= 1;
+        }
+        else {
+            top();
+        }
+    }
+
+    public void down() {
+        if (selected + 1 < selectable.size()) {
+            selected += 1;
+        }
+    }
+
+    public void top() {
+        if (selectable.size() > 0) {
+            selected = 0;
```

Comment:
This seems to stop the selection from going into negative index, but perhaps we want to instead set it to the last element instead?  (So it has wrap-around selection)?  And a similar change for     public void down().

Reply:
I personally dislike wrap-around selection most of the time. I think it can be more useful for longer lists, but for 2-3 items seems a bit silly.